src/core/component.ts

```diff
@@ -0,0 +1,3 @@
+export interface Component {
+    Process(input: any): any;
+}
```

Comment:
Yes thats true. 

My thougths behind this interface was, to force the developer to use a consistent pattern to write a Component. Also maybe use a [decorator](https://www.typescriptlang.org/docs/handbook/decorators.html) in the component so we would not need to call the Component from *.mapper.ts, it would do it automaticly because of the decorator. But I'm really sceptical how good this would work out, to be honest.